aptos-move/aptos-e2e-comparison-testing/src/execution.rs

```diff
@@ -191,163 +252,219 @@ impl Execution {
-                    features,
-                ));
-            } else if let TransactionPayload::Multisig(multi_sig) = payload {
-                assert!(multi_sig.transaction_payload.is_some());
-                println!("Multisig transaction is not supported yet");
+                let enable_v7 = |features: &mut Features| {
+                    if v2_flag {
+                        features.enable(FeatureFlag::VM_BINARY_FORMAT_V7);
+                    } else {
+                        features.enable(FeatureFlag::VM_BINARY_FORMAT_V6);
+                    }
+                };
+                if let Some(debugger) = debugger_opt {
+                    let data_view =
+                        DataStateView::new(debugger, version, executor.data_store().clone());
+                    let mut features =
+                        Features::fetch_config(&data_view.as_move_resolver()).unwrap_or_default();
+                    enable_v7(&mut features);
+                    return Some(executor.try_exec_entry_with_resolver(
+                        senders,
+                        entry_function,
+                        &data_view.as_move_resolver(),
+                        features,
+                    ));
+                } else {
+                    let mut features =
+                        Features::fetch_config(&executor.data_store().clone().as_move_resolver())
+                            .unwrap_or_default();
+                    enable_v7(&mut features);
+                    return Some(executor.try_exec_entry_with_resolver(
+                        senders,
+                        entry_function,
+                        &executor.data_store().clone().as_move_resolver(),
+                        features,
+                    ));
+                }
             }
         }
-        None
+        if let Some(debugger) = debugger_opt {
+            let data_view = DataStateView::new(debugger, version, executor.data_store().clone());
+            Some(
+                executor
+                    .execute_transaction_block_with_resolver([txn.clone()].to_vec(), &data_view)
+                    .map(|res| res[0].clone().into()),
+            )
+        } else {
+            Some(
+                executor
+                    .execute_transaction_block(vec![txn.clone()])
+                    .map(|res| res[0].clone().into()),
+            )
+        }
     }
 
     fn print_mismatches(
+        &self,
         cur_version: u64,
         res_1: &Result<(WriteSet, Vec<ContractEvent>), VMStatus>,
         res_2: &Result<(WriteSet, Vec<ContractEvent>), VMStatus>,
     ) {
         match (res_1, res_2) {
             (Err(e1), Err(e2)) => {
-                if e1 != e2 {
-                    println!("error is different at {}", cur_version);
-                    println!("error {} is raised from V1", e1);
-                    println!("error {} is raised from V2", e2);
+                if e1.message() != e2.message() {
```

Comment:
Should we be comparing status codes as well?

Reply:
done